src/Features/CSharp/Portable/CodeRefactorings/InvertIf/InvertIfCodeRefactoringProvider.Analyzer.cs

```diff
@@ -0,0 +1,365 @@
+                }
+            }
+
+            protected override TextSpan GetHeaderSpan(IfStatementSyntax ifStatement)
+            {
+                return TextSpan.FromBounds(
+                    ifStatement.IfKeyword.SpanStart,
+                    ifStatement.CloseParenToken.Span.End);
+            }
+
+            protected override bool IsElselessIfStatement(IfStatementSyntax ifStatement)
+            {
+                return ifStatement.Else == null;
+            }
+
+            protected override bool CanInvert(IfStatementSyntax ifStatement)
+            {
+                return ifStatement.IsParentKind(SyntaxKind.Block, SyntaxKind.SwitchSection);
+            }
+
+            protected override int GetNearmostParentJumpStatementRawKind(IfStatementSyntax ifStatement)
+            {
+                foreach (var node in ifStatement.Ancestors())
+                {
+                    switch (node.Kind())
+                    {
+                        case SyntaxKind.SwitchSection:
+                            return (int)SyntaxKind.BreakStatement;
+
+                        case SyntaxKind.LocalFunctionStatement:
+                        case SyntaxKind.SetAccessorDeclaration:
+                        case SyntaxKind.GetAccessorDeclaration:
+                        case SyntaxKind.AddAccessorDeclaration:
+                        case SyntaxKind.RemoveAccessorDeclaration:
+                        case SyntaxKind.MethodDeclaration:
+                        case SyntaxKind.ConstructorDeclaration:
+                        case SyntaxKind.DestructorDeclaration:
+                        case SyntaxKind.OperatorDeclaration:
+                        case SyntaxKind.ConversionOperatorDeclaration:
+                        case SyntaxKind.AnonymousMethodExpression:
+                        case SyntaxKind.SimpleLambdaExpression:
+                        case SyntaxKind.ParenthesizedLambdaExpression:
+                            return (int)SyntaxKind.ReturnStatement;
+
+                        case SyntaxKind.DoStatement:
+                        case SyntaxKind.WhileStatement:
+                        case SyntaxKind.ForStatement:
+                        case SyntaxKind.ForEachStatement:
+                        case SyntaxKind.ForEachVariableStatement:
+                            return (int)SyntaxKind.ContinueStatement;
+                    }
+                }
+
+                return -1;
+            }
+
+            protected override (SyntaxNode first, SyntaxNode last) GetIfBodyStatementRange(IfStatementSyntax ifStatement)
+            {
+                var statement = ifStatement.Statement;
+                return (statement, statement);
+            }
+
+            protected override IEnumerable<(SyntaxNode first, SyntaxNode last)> GetSubsequentStatementRange(IfStatementSyntax ifStatement)
+            {
+                StatementSyntax innerStatement = ifStatement;
+                foreach (var node in ifStatement.Ancestors())
```

Comment:
needs to def explain what it's doing and why it keeps walking up.

Reply:
(pending)